modules/db/src/test/java/org/mule/module/db/internal/domain/connection/SimpleConnectionFactoryTestCase.java

```diff
@@ -44,6 +49,22 @@ public void failsOnConnectionError() throws Exception
 
         connectionFactory.create(dataSource);
     }
+    
+    @Test(expected = ConnectionCreationException.class)
+    public void failsOnConnectionThenLogWithoutCredentials() throws Exception
+    {
+        when(dataSource.getConnection()).thenThrow(new RuntimeException(EXCEPTION_MESSAGE));
+
+        try
+        {
+            connectionFactory.create(dataSource);
```

Comment:
It fails in case no exception is thrown because there is an expected ConnectionCreationException using an annotation in the method.